docs/source/api/top-level-await.md

```diff
@@ -0,0 +1,120 @@
+
+Top level await is only enabled by default on the server. You can enable it for the client by setting the env var `METEOR_ENABLE_CLIENT_TOP_LEVEL_AWAIT` to `true`. There are a couple known issues with using TLA on the client:
+
+1. It breaks any files in `/client/compatibility` since it now wraps those files in a function
+2. Hot module replacement has not been updated to work with TLA
+
+### What is an async module?
+
+With top level await, some modules are considered async, which affects how it behaves. There are two ways a module can become an async module:
+1. It uses top level await
+2. It imports a module that is async
+
+For example, this module (`setup.js`) would be async because it uses top level await:
+
+```js
+await setupLanguages();
+```
+
+This module (`main.js`) would be sync:
+
+```js
+console.log('in main.js');
+```
+
+However, if it imports `setup.js` which does use top level await, then `main.js` also becomes async.
+
+```js
+import './setup.js';
+
+console.log('in main.js');
+```
+
+### `require`
+
+When using `require` to load an async module, instead of directly returning a module's exports, it will return a promise that resolves to the module's exports.
+
+```js
+// resolves to the exports of init.js
+const promise = require('./init.js');
+```
+
+If you are using `require`, this does mean you need to be somewhat careful when adding/removing top level await in a file since you also have to update where the module is required.
+Since a module becomes async if it depends on an async module, this could affect more than just the individual modules using top level await.
+
+When possible, you can use ecmascript import syntax or dynamic imports instead so you don't have to worry about which modules are sync or async.
+
+### Nested imports
+
+For nested imports, it refers to using `import ...` outside of the root of a module, for example in an if block or a function.
+
+```js
+if (Meteor.isClient) {
+  import './init-client.js';
+}
+
+export function showNotification(message) {
+  import show from './notifications.js';
+
+  show(message);
+}
+```
+
+ This is a feature unique to Meteor, so the top level await specification wasn't written to work with nested imports. Using nested imports to import a sync module continues to work, but it will throw an error if used to import an async module. You can use `require` or dynamic imports for async modules in these situations.
+
+### Using in packages
+
+Top level await is only supported started in Meteor 3.  Build plugins are able to use top level await in older Meteor versions since the runtime is bundled when they are published.
+
+If you want to ensure your package only runs in versions of Meteor that support top level await, you can have your package use `isobuild:top-level-await`:
+
+```js
+Package.onUse(function (api) {
+  // Do not allow this package to be used in pre-Meteor 3 apps.
```

Comment:
Wouldn't `api.versionsFrom('3.0') have the same effect?

Reply:
Not always. api.versionsFrom is just an easier way to provide the versions for any core packages that the package uses. If the package doesn't use any core packages, or the core packages it does use have the same version as the previous Meteor release, then it would allow the package to run in older Meteor versions.